test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/BundleEdgeCaseTests.cs

```diff
@@ -91,5 +96,139 @@ public async Task GivenABundleWithConditionalUpdateByReference_WhenExecutedWithM
                 Assert.Equal("2", bundleResponse2.Resource.Entry[i].Resource.Meta.VersionId);
             }
         }
+
+        [Fact]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]
```

Comment:
Why is this only for SQL? The logic looks like it would work for Cosmos as well.

Reply:
That's a good question. I created this test focused on a bug that only happened in Gen2 due the way how MergeAsync works. But it makes sense to keep it running for Gen1 and Gen2.